src/js/popup.js

```diff
@@ -339,6 +342,97 @@ $(document).ready(function () {
         });
     });
 
+    $('#customdomains').click(function () {
+        dmUIClick('customdomains');
+        if (!$(this).hasClass('disabled')) {
+            $('#customDomainList').toggle();
+            $('main').toggle();
+        } else {
+            Materialize.toast(disabledOptionToast, 2000, 'toastNotification');
+        }
+    });
+
+    $('#backInternal').click(function () {
+        dmUIClick('backInternal');
+        $('#customDomainList').toggle();
+        $('main').toggle();
+    });
+
+    $('#clearInternal').click(function () {
+        dmUIClick('clearInternal');
+        // remove all added domains:
+        chrome.runtime.sendMessage({
+            JDTIfunction: 'removeDomain',
+            domainName: '',
+            removeAll: true
+        }, function (response) {
+            if (response.status === 'success') {
+                loadTemplateEditor();
+                Materialize.toast('Domains successfully removed', 2000, 'toastNotification');
+            } else {
+                if (response.message) {
+                    dmError('clearInternal', response.message);
+                    Materialize.toast(response.message, 2000, 'toastNotification');
+                } else {
+                    dmError('clearInternal', 'generic');
+                    Materialize.toast('Something went wrong. Please try again.', 2000, 'toastNotification');
+                }
+            }
+        });
+    });
+
+    $('#CustomDomainInput').keyup(function (event) {
+        if (event.keyCode === 13) {
+            $('#CustomDomainInputButton').click();
+        }
+    });
+
+    $('#CustomDomainInputButton').click(function () {
+        dmUIClick('CustomDomainInputButton');
+        var domainName = $('#CustomDomainInput').val();
+        $('#CustomDomainInput').val('');
```

Comment:
Tested, and the template editor reload does not clear the value, so moving this to the success block is necessary.